### PR TITLE
Update to Rust 1.71.1 (cargo CVE-2023-38497)

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

Was about time to update (cargo-deny was complaining), and there is a CVE in cargo, so seemed like the sensible thing to do.

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

https://blog.rust-lang.org/2023/08/03/cve-2023-38497.html

### CODE REVIEW

**Comment by markmandel** [comment]
Sod, have to wait for `joseluisq/rust-linux-darwin-builder:1.71.1` to be released.

**Comment by markmandel** [comment]
https://github.com/joseluisq/rust-linux-darwin-builder/issues/24 just for tracking.

File: src/endpoint/locality.rs [hunk 1]
```diff
@@ -225,7 +225,7 @@ impl LocalitySet {
 
     /// Inserts a new locality of endpoints.
     pub fn insert(&mut self, mut locality: LocalityEndpoints) {
-        let mut entry = self.0.entry(locality.locality.clone()).or_default();
+        let entry = self.0.entry(locality.locality.clone()).or_default();
```
Comment (XAMPPRocky) [inline src/endpoint/locality.rs:228]: How does this not need to be mutable? 🤔

**Review by XAMPPRocky — APPROVED** [review]

File: src/endpoint/locality.rs (hunk 1 above)
Comment (markmandel) [reply src/endpoint/locality.rs:228]: I just do what `cargo check` tell me?

**Pull request merged**